scripts/purge_queue.py

```diff
@@ -0,0 +1,128 @@
+import sys
+
+import argparse
+import boto3
+import json
+import more_itertools
+from azul import config
+import logging
+
+from urllib.parse import urlparse
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(description="Extract and or remove messages from SQS queues")
+
+    subparsers = parser.add_subparsers(help="sub-command help", dest="command")
+
+    parser_dump = subparsers.add_parser("dump", help="Dump contents of queue into designated file")
+    parser_dump.add_argument("queue_name", help="Name of the queue to obtain the messages from")
+    parser_dump.add_argument("file_name", help="Name of file to write output to")
+    parser_dump.add_argument("--delete", "-D", action='store_true',
+                             help="Purge messages from the queue after writing them to the specified file. "
+                                  "By default the messages will be returned to the queue")
+    parser_dump.add_argument("--json-body", "-J", action="store_true", help="Serialize json for queue message body")
+    subparsers.add_parser("list", help="List SQS queues in current deployment")
+
+    args = parser.parse_args(argv)
+    if args.command == "list":
+        list_queues()
+    elif args.command == "dump":
+        get_messages_from_queue(options=args)
+    else:
+        parser.parse_args(['--help'])
+
+
+def list_queues():
+    sqs = boto3.resource("sqs")
+    client = boto3.client("sqs")
+    all_queues = [q.url for q in sqs.queues.all()]
+    print('\n{:<35s}{:^25s}{:^20s}{:^20s}\n'.format("Queue Name",
+                                                    'Messages Available',
+                                                    'Messages in Flight',
+                                                    'Messages Delayed'))
+    for queue_url in all_queues:
+        _, _, queue_name = urlparse(queue_url).path.rpartition("/")
+        azul_queue_name = config.is_azul_queue(queue_name)
+        if azul_queue_name:
+            queue = client.get_queue_attributes(
+                QueueUrl=queue_url,
+                AttributeNames=['ApproximateNumberOfMessages',
+                                'ApproximateNumberOfMessagesNotVisible',
+                                'ApproximateNumberOfMessagesDelayed'])
+            queue_attributes = queue['Attributes']
+            print('{:<35s}{:^25s}{:^20s}{:^20s}'.format(queue_name,
+                                                        queue_attributes['ApproximateNumberOfMessages'],
+                                                        queue_attributes["ApproximateNumberOfMessagesNotVisible"],
+                                                        queue_attributes["ApproximateNumberOfMessagesDelayed"]))
+
+
+def get_messages_from_queue(options):
+    sqs = boto3.resource("sqs")
+    logger = logging.getLogger(__name__)
+    queue_name = sqs.get_queue_by_name(QueueName=options.queue_name).url
+    output_file_name = options.file_name
+    queue = sqs.Queue(url=queue_name)
```

Comment:
Why? You already had the Queue instance when you did `sqs.get_queue_by_name(QueueName=options.queue_name)` above.